GitHub Actions: check lesson template

maxim-belkin opened the pull request:
Action to dry-run changes we're making here to this repository on a number of lessons. I chose:
- swcarpentry/shell-novice
- datacarpentry/r-intro-geospatial
- librarycarpentry/lc-git

Tests are performed on all three operating systems: macos, linux, and windows (git for bash)

maxim-belkin committed:
GitHub Actions: check lesson template

fmichonneau committed:
.github/workflows/template.yml: refactor R workflow

maxim-belkin commented on .github/workflows/template.yml:
```diff
@@ -0,0 +1,96 @@
+        with:
+          r-version: 'release'
+
+      - name: Install R packages
+        if: steps.check-rmd.outputs.count != 0
+        run: |
+          install.packages('remotes', 'rprojroot', 'renv')
+        shell: Rscript {0}
+
+      - name: Cache R packages
+        if: runner.os != 'Windows' && steps.check-rmd.outputs.count != 0
+        uses: actions/cache@v1
+        with:
+          path: ${{ env.R_LIBS_USER }}
+          key: ${{ runner.os }}-${{ hashFiles('.github/R-version') }}-1-${{ hashFiles('.github/depends.Rds') }}
+          restore-keys: ${{ runner.os }}-${{ hashFiles('.github/R-version') }}-1-
```
@fmichonneau, I believe we have to swap the order of "Cache R packages" and "Install R packages" and check for cache hits (`steps.cache-r.outputs.cache-hit != 'true'`).

fmichonneau replied:
these are the packages that are needed for all the R-based lessons. The idea is that these packages would always be installed fresh but the packages that are lesson-specific get cached.

maxim-belkin replied:
Currently `Cache R packages` step is not used -- the idea is to skip package installation step when `steps.cache-r.outputs.cache-hit == 'true`.

Don't we also need 'knitr', 'stringr', 'checkpoint', 'ggplot2'?

fmichonneau replied:
not anymore, we now use a combination of `renv` + `rprojroot` which takes care of identifying all the dependencies in our lessons that are then installed as part of the build process.

maxim-belkin replied:
> not anymore, we now use a combination of `renv` + `rprojroot` which takes care of identifying all the dependencies in our lessons that are then installed as part of the build process.

Great!

So, right now the caching step is used to cache downloads only and that's exactly what you want. I'm 👍 with that.

maxim-belkin replied:
wait, I think current order has to be switched, though -- "Install R packages" step won't see any cache because cache is created only after "Install R packages" is executed. Is there a way to download R packages before installing them?

maxim-belkin replied:
we can have `cache-restore` and `cache-save` steps before and after we install R packages.

maxim-belkin replied:
Resolved in https://github.com/maxim-belkin/styles/pull/3

fmichonneau committed:
rework installation of lesson dependencies

fmichonneau committed:
fix typo

Co-authored-by: Maxim Belkin <maxim.belkin@gmail.com>

fmichonneau committed:
remove sudo

Co-authored-by: Maxim Belkin <maxim.belkin@gmail.com>

fmichonneau committed:
change ubuntu version

Co-authored-by: Maxim Belkin <maxim.belkin@gmail.com>

fmichonneau committed:
specify that it is an R-specific command

fmichonneau committed:
add missing line to cache current and available package versions

fmichonneau committed:
add install-rmd-deps as PHONY target

fmichonneau committed:
suppress makefile output when running mkdir -p

fmichonneau committed:
specify that steps only applies to R-based lessons

maxim-belkin committed:
GitHub Actions: template: fix for pull requests

Avoid `pull_request: []` warning

    Invalid type found: `object` was expected but an array was found
